### PR TITLE
Make Activation Tabs Closable

### LINKED ISSUE #6271: Slice 6: Make Mod Activation Panels Closable
- Add a `CloseButton` to the modActivationPanel `TabWithDivider`
    - Update `Tabs.tsx:onClosePanel` to check if the panel is a modActivationPanel
        - dispatch the `hideModActivationPanel` sidebar action

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Closes #6271 
- Some refactoring
- Fix default tab bug

## Demo

- [Loom](https://www.loom.com/share/71f486b12c9940d1b39572fc71f1d4c8?sid=c0617687-5307-4ed8-9884-12788bb5b55d)

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/sidebar/activateRecipe/ActivateMultipleModsPanel.tsx [hunk 1]
```diff
@@ -46,15 +43,9 @@ const MultipleSuccessPanel: React.FC<{ results: ModResultPair[] }> = ({
...
   const reduxDispatch = useDispatch();
-  const sidebarHasModPanels = useSelector(selectSidebarHasModPanels);
 
   async function handleActivationDecision() {
     reduxDispatch(sidebarSlice.actions.hideModActivationPanel());
-
-    if (!sidebarHasModPanels) {
-      const topFrame = await getTopLevelFrame();
-      void hideSidebar(topFrame);
-    }
```
Comment (turbochef) [inline src/sidebar/activateRecipe/ActivateMultipleModsPanel.tsx:57]: This is already being handle in `useHideEmptySidebar.ts`.

File: src/sidebar/activateRecipe/ActivateModInputs.tsx [hunk 2]
```diff
@@ -157,18 +157,6 @@ const ActivateModInputs: React.FC<ActivateRecipeInputsProps> = ({
...
-          variant="outline-danger"
-          onClick={() => {
-            reportEvent(Events.MOD_ACTIVATION_CANCEL, {
-              recipeId: recipe?.metadata?.id,
-            });
-            onClickCancel();
-          }}
-        >
-          Cancel
-        </Button>
```
Comment (turbochef) [inline src/sidebar/activateRecipe/ActivateModInputs.tsx:171]: Waiting on @BrandonPxBx to weigh in on whether we want to keep this.

File: src/sidebar/sidebarSelectors.test.ts
Comment (turbochef) [inline src/sidebar/sidebarSelectors.test.ts:1]: only `selectSidebarHasModPanels ` was being tested here. I removed `selectSidebarHasModPanels `.

File: src/sidebar/activateRecipe/ActivateModInputs.tsx (hunk 2 above)
Comment (turbochef) [reply src/sidebar/activateRecipe/ActivateModInputs.tsx]: decided to keep it

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**